### ISSUE TITLE
create_callbacks() crash under Python 2.7 if snapshots

### ISSUE THREAD

**Issue by pderian** [fizyr/keras-retinanet#345]
In `train.py:create_calbacks()` l. 99: `exist_ok` is valid option of `os.makedirs()` under Python 3, not Python 2.7. Tests pass as they have the` --no-snapshot` option.

https://github.com/fizyr/keras-retinanet/blob/7ba5ec2bf3497c9fd4c8dd2534d0d1d15e717a9b/keras_retinanet/bin/train.py#L99

**Comment by hgaiser** [comment]
Nice catch! Can you make a PR to fix this? Probably it's easiest to wrap it in an `if os.path.isdir(...)`

**Comment by de-vri-es** [comment]
The proper way to deal with existing directories is to just try to create it and catch the error. Checking and then creating has a small race condition.

**Comment by hgaiser** [comment]
Fair enough, let's go that way.

**Comment by de-vri-es** [comment]
To be 100% correct we should check the file type if it exists already. If an error is raised and the path is not a directory, the error shouldn't be ignored.

**Comment by pderian** [comment]
Something along the lines of:
```
    # save the prediction model
    if args.snapshots:
        # ensure directory created first; otherwise h5py will error after epoch.
        try:
            os.makedirs(args.snapshot_path)
        # if an error is raised
        except OSError as err_makedirs:
            # and it is NOT because the dir exists already
            if not os.path.isdir(args.snapshot_path):
                raise(err_makedirs)
            # otherwise we're good (following the previous behavior)
```
That is to say: if `os.makedirs()` failed because the dir exists already, it is fine (mimics `exist_ok=True`), but if it failed and `args.snapshot_path` is not a dir then the error propagates.

**Comment by de-vri-es** [comment]
Pretty much. One thing though: the best way to re-raise a caught exception is to simply invoke `raise` without arguments. That way the stack trace of the original exception is preserved (though the new raise location is also added, I think).

It would also be nice to add it as a utility function. It is fine to define it locally in the file where we need it though. If we need it in other places too we can move it to `utils` later.

**Comment by de-vri-es** [comment]
Fixed by #350. Thanks again for your contribution :)
